src/celengine/timeline.cpp

```diff
@@ -46,14 +44,13 @@ Timeline::appendPhase(TimelinePhase* phase)
             return false;
     }
 
-    phase->addRef();
-    phases.push_back(phase);
+    phases.push_back(shared_ptr<TimelinePhase>(phase));
```

Comment:
have you tried to leave `phases.push_back(phase);`? compiler should generate calls to create shared_ptr

Reply:
I'm nearly sure 've tried, also hoping compiler will do the rest.